### ISSUE TITLE
partition the customers based on how many coupons they bought

### ISSUE THREAD

**Issue by daxiongshu** [daxiongshu/Coupon-Purchase-Prediction#3]
Hi, I'm basically done with my experiment. Please see the code that I updated.

The idea is that partition the customers into two categories:

1) frequent coupon buyers, which is defined as the one who bought more than 30 different coupons (COUPON_HASH_id). I actually think the optimal threshold should be higher than 30. I'll figure out that

2) others

So I think for 1), a per customer model will do better like what we saw in eeg contest.

I used our previous best model, ave3sub.csv as the baseline and updates predictions of the "frequent coupon buyers" with my per customer model. For example, in ave3sub.csv and sub3.csv (0.0109 lb), there are only 310 rows different, which stand for the frequent buyers that I update with my model. Please note that there are about 800 users who bought more than 30 coupons, so the 310 users above is just a subset of them. I chose only 310 because I thought a threshold of 45 something may be better and I saw the moving average of coupons go below 45 after these 310 users so I stop it. Anyway we could definitely improve this with a more systematic way. 

One thing you could try is to reproduce sub10.csv, but remove those 310 users from both train set and test set. Let's say these 310 users are outliers for other (type 2) customers.

By the way, could you send me a link of xgb 0.3, I didn't find it in email or github. Thank you very much!

Best regards,

**Comment by SudalaiRajkumar** [comment]
Here is the link for xgboost
https://github.com/dmlc/xgboost/releases/tag/v0.32

**Comment by SudalaiRajkumar** [comment]
Okay. I will try to reproduce sub10 without these customers. Do you have a list of these customers? 
I could be able to get them if it is not readily available. Thank you.

**Comment by daxiongshu** [comment]
Thank you very much.

Sure, I will upload the list in a minute.

**Comment by SudalaiRajkumar** [comment]
Thank you very much Carl

**Comment by daxiongshu** [comment]
Please check the list.csv in my folder. No problem. Your work lay a great foundation for us! : )

**Comment by SudalaiRajkumar** [comment]
Thank you

**Comment by SudalaiRajkumar** [comment]
Sorry I closed this by mistake. Thank you for the LinkedIn endorsements :)
